code/modules/vehicles/mecha/combat/justice.dm

```diff
@@ -0,0 +1,455 @@
+	if(prob(DISMEMBER_CUALITY_HIGTH))
+		cut_bodypart.dismember(BRUTE)
+
+
+/obj/vehicle/sealed/mecha/justice/mob_exit(mob/M, silent, randomstep, forced)
+	. = ..()
+	if(!stealth_action?.on)
+		return
+	animate(src, alpha = 255, time = 0.5 SECONDS)
+	playsound(src, 'sound/mecha/mech_stealth_effect.ogg' , 75, FALSE)
+
+/obj/vehicle/sealed/mecha/justice/Bump(atom/obstacle)
+	. = ..()
+	if(!isliving(obstacle))
+		return
+	if(!stealth_action?.on)
+		return
+	animate(src, alpha = 255, time = 0.5 SECONDS)
+	playsound(src, 'sound/mecha/mech_stealth_effect.ogg' , 75, FALSE)
+
+/obj/vehicle/sealed/mecha/justice/Bumped(atom/movable/bumped_atom)
+	. = ..()
+	if(!isliving(bumped_atom))
+		return
+	if(!stealth_action?.on)
+		return
+	animate(src, alpha = 255, time = 0.5 SECONDS)
+	playsound(src, 'sound/mecha/mech_stealth_effect.ogg' , 75, FALSE)
+
+/obj/vehicle/sealed/mecha/justice/take_damage(damage_amount, damage_type, damage_flag, sound_effect, attack_dir, armour_penetration)
+	if(stealth_action?.on)
+		animate(src, alpha = 255, time = 0.5 SECONDS)
+		playsound(src, 'sound/mecha/mech_stealth_effect.ogg' , 75, FALSE)
+	if(LAZYLEN(occupants))
+		if(prob(60))
+			new /obj/effect/temp_visual/mech_sparks(get_turf(src))
+			playsound(src, 'sound/mecha/mech_stealth_effect.ogg' , 75, FALSE)
+			return
+	return ..()
+
+/datum/action/vehicle/sealed/mecha/invisibility
+	name = "Invisibility"
+	button_icon_state = "mech_stealth_off"
+	/// Is invisibility activated.
+	var/on = FALSE
+	/// Recharge check.
+	var/charge = TRUE
+	/// Block movement if we start aoe attack from invisibility.
+	var/start_attack = FALSE
+	/// Aoe pre attack sound.
+	var/stealth_pre_attack_sound = 'sound/mecha/mech_stealth_pre_attack.ogg'
+	/// Aoe attack sound.
+	var/stealth_attack_sound = 'sound/mecha/mech_stealth_attack.ogg'
+
+/datum/action/vehicle/sealed/mecha/invisibility/Trigger(trigger_flags)
+	. = ..()
+	if(chassis.weapons_safety)
+		owner.balloon_alert(owner, "safty is on!")
+		return
+	if(!charge)
+		owner.balloon_alert(owner, "on recharge!")
+		return
+	new /obj/effect/temp_visual/mech_sparks(get_turf(chassis))
+	on = !on
+	playsound(chassis, 'sound/mecha/mech_stealth_effect.ogg' , 75, FALSE)
+	if(on)
+		for(var/mob/living/occupant in chassis.occupants)
+			var/datum/action/vehicle/sealed/mecha/charge_attack/charge_action = LAZYACCESSASSOC(chassis.occupant_actions, occupant, /datum/action/vehicle/sealed/mecha/charge_attack)
+			if(charge_action?.on)
+				charge_action.on = FALSE
+		animate(chassis, alpha = 0, time = 0.5 SECONDS)
+		button_icon_state = "mech_stealth_on"
+		addtimer(CALLBACK(src, PROC_REF(end_stealth)), 20 SECONDS)
+		RegisterSignal(chassis, COMSIG_MECHA_MELEE_CLICK, PROC_REF(stealth_attack_aoe))
+	else
+		addtimer(CALLBACK(src, PROC_REF(charge)), 5 SECONDS)
+		animate(chassis, alpha = 255, time = 0.5 SECONDS)
+		button_icon_state = "mech_stealth_off"
+		UnregisterSignal(chassis, COMSIG_MECHA_MELEE_CLICK)
```

Comment:
You do this in a bunch of places, make it into a proc instead of copy/pasting the code block all over the place.